src/test/run-pass/arbitrary_enum_discriminant.rs

```diff
@@ -0,0 +1,55 @@
+#![feature(arbitrary_enum_discriminant, const_raw_ptr_deref, test)]
```

Comment:
I would prefer to see all of these tests, including the feature gate tests, moved into `src/test/ui/enum-discriminant/*` for better organization and discoverability when it is time for stabilization.